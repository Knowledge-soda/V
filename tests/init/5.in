:a, b
--(var) := =(var, +(v(var), -1))
++(var) := =(var, +(v(var), 1))
+=(var, n) := =(var, +(v(var), n))
sum(x) := v(=(b, 0) =(a, 0) rep: ++(b) +=(a, v(b)) ijmp(<(v(b), x), rep) a)
