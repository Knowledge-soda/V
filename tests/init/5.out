_start:
    NAME(id) BEGIN END
--:
    BEGIN NAME(var) END : NAME(=) BEGIN NAME(var) SEP NAME(+) BEGIN NAME(v) BEGIN NAME(var) END SEP NUM(-1) END END
++:
    BEGIN NAME(var) END : NAME(=) BEGIN NAME(var) SEP NAME(+) BEGIN NAME(v) BEGIN NAME(var) END SEP NUM(1) END END
+=:
    BEGIN NAME(var) SEP NAME(n) END : NAME(=) BEGIN NAME(var) SEP NAME(+) BEGIN NAME(v) BEGIN NAME(var) END SEP NAME(n) END END
sum:
    BEGIN NAME(x) END : NAME(v) BEGIN NAME(=) BEGIN NAME(b) SEP NUM(0) END NAME(=) BEGIN NAME(a) SEP NUM(0) END PLACE(rep) NAME(++) BEGIN NAME(b) END NAME(+=) BEGIN NAME(a) SEP NAME(v) BEGIN NAME(b) END END NAME(ijmp) BEGIN NAME(<) BEGIN NAME(v) BEGIN NAME(b) END SEP NAME(x) END SEP NAME(rep) END NAME(a) END
